core/native_type_traits.h

```diff
@@ -133,4 +133,34 @@ struct NativeTypeTraits<IDLString> : public NativeTypeTraitsBase<IDLString> {
   }
 };
 
+// FIXME(Hwansung): should be generated automatically
+#include "examples/operation_type.h"
+template <>
+struct NativeTypeTraits<IDLOperationType>
+    : public NativeTypeTraitsBase<IDLOperationType> {
+  static OperationType NativeValue(const Napi::Env& env,
+                                   const Napi::Value& js_value) {
+    if (!js_value.IsString()) {
+      Napi::TypeError::New(env, "It's an invalid string.")
+          .ThrowAsJavaScriptException();
+      return OperationType::NONE;
+    }
+
+    std::string value = js_value.ToString().Utf8Value();
+    if (value.compare("add") == 0) {
+      return OperationType::ADD;
+    } else if (value.compare("sub") == 0) {
+      return OperationType::SUB;
+    } else if (value.compare("mul") == 0) {
+      return OperationType::MUL;
+    } else if (value.compare("div") == 0) {
+      return OperationType::DIV;
+    }
+
+    Napi::TypeError::New(env, "It's an invalid enum value.")
+        .ThrowAsJavaScriptException();
+    return OperationType::NONE;
+  }
+};
```

Comment:
If my PR (https://github.com/lunchclass/bacardi/pull/108) is merged before this PR, 
This Type must be implement IsTypeEquals methods like below : 

```typescript
static bool IsTypeEquals(const Napi::Value& js_value) {
    return js_value.IsOperationType();
}
```

otherwise, i should implement this method! :)

Reply:
okay i will rebase it.
unfortunately we should implement more things in IsTypeEquals function instead of to use ``` js_value.IsOperationType()```.
because js_value will be string type in this case. and it should be check that value can be Enum type.

Reply:
If ```IsOperationType()``` implementation is too large to apply this PR, I think the implementation can be divided another PR.

Other things LGTM.

Reply:
If you complete implement ation```IsOperationType()``` .

you can check argument first, and yiu can return value.

please refer to below code :